cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -609,13 +612,22 @@ private List<ApplicationId> decodeAndValidateBatchApplication(NamespaceId namesp
           throw new BadRequestException("Missing 'appId' in the request element.");
         }
         String appId = obj.get("appId").getAsString();
+        String version = null;
 
-        JsonElement version = obj.get("version");
-        if (version == null) {
-          result.add(validateApplicationId(namespaceId, appId));
+        JsonElement versionObj = obj.get("version");
+        if (versionObj == null) {
+          // get the latest version of the app instead of -SNAPSHOT
+          try {
+            ApplicationDetail appLatestVersionDetail = applicationLifecycleService.getAppLatestVersionDetail(
```

Comment:
this is a lot of extra I/O and shouldn't be done. We shouldn't be making db calls when decoding user input, we should just be decoding whatever the user sent. Later on, when the caller is using the returned list, they can decide what they want to do if an app id is invalid.